recipes_source/generic_join.rst

```diff
@@ -0,0 +1,434 @@
+Distributed Training with Uneven Inputs Using the Join Context Manager
```

Comment:
This page might better fit in the tutorial folder than a recipe (which is usually short and simple), as this one has quite a bit of content.

I would suggest add this to `https://pytorch.org/tutorials/advanced/` and add a link to the overview page under the DDP section: https://pytorch.org/tutorials/beginner/dist_overview.html